Letter customization

SAUMILDHANKAR opened the pull request:
#428 #529

paramsiddharth reviewed (dismissed):
Looks good to me! Let's aim to have it merged tonight after a final review by @DietBepis1 .

DietBepis1 commented on src/components/LetterLoad.vue:
```diff
@@ -63,7 +70,9 @@ export default {
   data () {
     return {
-      isSubmitted: true
+      date: new Date().toLocaleString(),
```
I think we should perhaps pass `en-US` to ensure that the date is always formatted the same way. `new Date.toLocaleString('en-US')`

or alternatively, `new Intl.DateTimeFormat('en-US').format(new Date()))`

SAUMILDHANKAR replied:
Hello Glenn, I have updated the date format using:
```new Intl.DateTimeFormat('en-US').format(new Date())```

DietBepis1 commented on src/components/LetterLoad.vue:
```diff
@@ -83,6 +92,9 @@ export default {
 
       return ''
     }
+  },
```
I think we could use a watcher here for great effect. I started to port this code over and did something like this:

```
watch: {
    customMessage(newMessage) {
      console.log(newMessage)
      this.$store.commit('setGenericValue', { key: 'userCustomMessage', value: newMessage })
    }
  }
```

The watcher will  then update the Vuex store everytime the customMessage variable changes.

DietBepis1 replied:
I also propose renaming the Vuex key to be slightly more explicit.

SAUMILDHANKAR replied:
I have added the watcher and renamed the key as suggested. Hope it is fine.

SAUMILDHANKAR commented on src/components/TakeAction.vue:
```diff
@@ -165,8 +174,10 @@ export default {
+  mounted() {},
   methods: {
     nextPage(attrs) {
+      this.$refs.letterLoad.persistCustomization()
```
@DietBepis1: While testing the custom message flow, I see an error 
```this.$refs.letterLoad.persistCustomization is not a function`` and I am reading up on how to fix this. Please let me know if you have any suggestions. Thank you.

DietBepis1 replied:
I think we can just get rid of this now that we are using the watcher.

SAUMILDHANKAR replied:
Removed it.

paramsiddharth commented:
@paramsiddharth I need to review it locally.

SAUMILDHANKAR commented:
@DietBepis1 Hi Glenn, just wanted to share the screen recording for the error that is coming up with current code.

https://user-images.githubusercontent.com/843538/232210484-b5d812a3-9acd-466d-84aa-e5060c1d24db.mp4

Also, did some analysis with help from @yoyoyojoe on ways to resolve the error. Essentially, we made two main changes:
  - Removed persistCustomization() function from LetterLoad.vue file and committed userCustomMessage within the watch option.
  - Removed this.$refs.letterLoad.persistCustomization() from TakeAction.vue file.
  
Please see the screen recording below to see the flow after these changes. Notice userCustomMessage getting added in this.$store.state

https://user-images.githubusercontent.com/843538/232214448-09934154-407b-4b46-a8ee-16e323355d7b.mp4

I can add a commit for you to see the code changes if this is something you would like me to proceed with. Please let me know if you would recommend another approach. Looking forward to your feedback. Thank you.

DietBepis1 commented:
> @DietBepis1 Hi Glenn, just wanted to share the screen recording for the error that is coming up with current code.
> 
> Also, did some analysis with help from @yoyoyojoe on ways to resolve the error. Essentially, we made two main changes:
> 
>     * Removed persistCustomization() function from LetterLoad.vue file and committed userCustomMessage within the watch option.
> 
>     * Removed this.$refs.letterLoad.persistCustomization() from TakeAction.vue file.
> 
> 
> Please see the screen recording below to see the flow after these changes. Notice userCustomMessage getting added in this.$store.state
> 
> I can add a commit for you to see the code changes if this is something you would like me to proceed with. Please let me know if you would recommend another approach. Looking forward to your feedback. Thank you.

@SAUMILDHANKAR and @yoyoyojoe, just wanted to say that this was excellent collaboration and a great demonstration. 👍🏽

DietBepis1 commented on src/components/LetterLoad.vue:
```diff
@@ -83,6 +92,15 @@ export default {
+      this.$store.commit('setGenericValue', { key: 'userCustomMessage', value: newMessage })
+    }
+  },
+  persistCustomization(){
```
Unless I am missing it somewhere, this is not used anywhere now and we can delete it.

SAUMILDHANKAR replied:
Thanks for checking this Glenn. I have deleted the persistCustomization() function.

DietBepis1 commented:
@paramsiddharth and @SAUMILDHANKAR, how did this PR get branched off of the original one from Sammy? The reason I am asking is that I noticed a missing prop `repName` error in the videos posted in the last few comments. That prop is removed in our main branch. The PR itself looks good and ready to merge, but I am worried that we may introduce bugs into the main codebase since I can see this unused prop.

paramsiddharth commented:
> @paramsiddharth and @SAUMILDHANKAR, how did this PR get branched off of the original one from Sammy? The reason I am asking is that I noticed a missing prop `repName` error in the videos posted in the last few comments. That prop is removed in our main branch. The PR itself looks good and ready to merge, but I am worried that we may introduce bugs into the main codebase since I can see this unused prop.

@DietBepis1 Saumil branched it off Sam's PR's branch into a branch in this repository – The prop must have been introduced while resolving the conflicts. It is something we'll have to pair up on, I believe. Perhaps outside the pair hours, some dedicated time.

SAUMILDHANKAR commented:
> > @paramsiddharth and @SAUMILDHANKAR, how did this PR get branched off of the original one from Sammy? The reason I am asking is that I noticed a missing prop `repName` error in the videos posted in the last few comments. That prop is removed in our main branch. The PR itself looks good and ready to merge, but I am worried that we may introduce bugs into the main codebase since I can see this unused prop.
> 
> @DietBepis1 Saumil branched it off Sam's PR's branch into a branch in this repository – The prop must have been introduced while resolving the conflicts. It is something we'll have to pair up on, I believe. Perhaps outside the pair hours, some dedicated time.

+1 to what Param just mentioned. Also, wanted to share that the workflow looked fine when I removed ```repName``` in my local environment and the error went away as well.

SAUMILDHANKAR commented on src/components/TakeAction.vue:
```diff
@@ -28,7 +28,12 @@
         <v-expansion-panel-content>
-          <letter-load :selected-rep="selectedRep" :letter-body="letterBody" />
+          <letter-load
+            ref="letterLoad"
```
I should have removed this line as well since it is no longer being referenced.
